pkg/xds/envoy/listeners/v3/access_log_configurer.go

```diff
@@ -0,0 +1,126 @@
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	"github.com/kumahq/kuma/pkg/envoy/accesslog"
+	"github.com/kumahq/kuma/pkg/util/proto"
+	"github.com/kumahq/kuma/pkg/xds/envoy"
+)
+
+const accessLogSink = "access_log_sink"
+
+type AccessLogConfigurer struct {
+	Mesh               string
+	TrafficDirection   envoy.TrafficDirection
+	SourceService      string
+	DestinationService string
+	Backend            *mesh_proto.LoggingBackend
+	Proxy              *core_xds.Proxy
+}
+
+func convertLoggingBackend(mesh string, trafficDirection envoy.TrafficDirection, sourceService string, destinationService string, backend *mesh_proto.LoggingBackend, proxy *core_xds.Proxy, defaultFormat string) (*envoy_accesslog.AccessLog, error) {
+	if backend == nil {
+		return nil, nil
+	}
+	formatString := defaultFormat
+	if backend.Format != "" {
+		formatString = backend.Format
+	}
+	format, err := accesslog.ParseFormat(formatString + "\n")
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid access log format string: %s", formatString)
+	}
+
+	variables := accesslog.InterpolationVariables{
+		accesslog.CMD_KUMA_SOURCE_ADDRESS:              net.JoinHostPort(proxy.Dataplane.GetIP(), "0"), // deprecated variable
+		accesslog.CMD_KUMA_SOURCE_ADDRESS_WITHOUT_PORT: proxy.Dataplane.GetIP(),                        // replacement variable
+		accesslog.CMD_KUMA_SOURCE_SERVICE:              sourceService,
+		accesslog.CMD_KUMA_DESTINATION_SERVICE:         destinationService,
+		accesslog.CMD_KUMA_MESH:                        mesh,
+		accesslog.CMD_KUMA_TRAFFIC_DIRECTION:           string(trafficDirection),
+	}
+
+	format, err = format.Interpolate(variables)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to interpolate access log format string with Kuma-specific variables: %s", formatString)
+	}
+
+	switch backend.GetType() {
+	case mesh_proto.LoggingFileType:
+		return fileAccessLog(format, backend.Conf)
+	case mesh_proto.LoggingTcpType:
+		return tcpAccessLog(format, backend.Conf)
+	default: // should be caught by validator
+		return nil, errors.Errorf("could not convert LoggingBackend of type %T to AccessLog", backend.GetType())
+	}
+}
+
+func tcpAccessLog(format *accesslog.AccessLogFormat, cfgStr *structpb.Struct) (*envoy_accesslog.AccessLog, error) {
+	cfg := mesh_proto.TcpLoggingBackendConfig{}
+	if err := proto.ToTyped(cfgStr, &cfg); err != nil {
+		return nil, errors.Wrap(err, "could not parse backend config")
+	}
+
+	httpGrpcAccessLog := &access_loggers_grpc.HttpGrpcAccessLogConfig{
+		CommonConfig: &access_loggers_grpc.CommonGrpcAccessLogConfig{
+			LogName: fmt.Sprintf("%s;%s", cfg.Address, format.String()),
+			GrpcService: &envoy_core.GrpcService{
+				TargetSpecifier: &envoy_core.GrpcService_EnvoyGrpc_{
+					EnvoyGrpc: &envoy_core.GrpcService_EnvoyGrpc{
+						ClusterName: accessLogSink,
+					},
+				},
+			},
+		},
+	}
+	// if err := format.ConfigureHttpLog(httpGrpcAccessLog); err != nil {
```

Comment:
HTTP Log is TODO?
The PR messages mention the TCP log.

Reply:
we use our custom grpcAccessLog server baked-in in kuma-dp to support sending logs via TCP